src/android/ListonicAds.java

```diff
@@ -0,0 +1,288 @@
+                    setAllParentsClip(parentView, false);
+                    rootView.addView(parentView);
+                }
+
+                parentView.bringToFront();
+                parentView.requestLayout();
+                parentView.requestFocus();
+            }
+        });
+    }
+
+    public static void setAllParentsClip(View v, boolean enabled) {
+        while (v.getParent() != null && v.getParent() instanceof ViewGroup) {
+            ViewGroup viewGroup = (ViewGroup) v.getParent();
+            viewGroup.setClipChildren(enabled);
+            viewGroup.setClipToPadding(enabled);
+            v = viewGroup;
+        }
+    }
+
+    private View getWebView(CordovaWebView webView) {
+        try {
+            return (View) webView.getClass().getMethod("getView").invoke(webView);
+        } catch (Exception e) {
+            return (View) webView;
+        }
+    }
+
+
+    private void prepare(final JSONObject options, final CallbackContext callbackContext) {
+        System.out.println("#debug ListonicAds prepare start");
+    }
+
+    private void setOptions(JSONObject options, CallbackContext callbackContext) {
+        if (options.has("width") && options.has("height")) {
+            cordovaInstance.getActivity().runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Integer width = null;
+                    Integer height = null;
+
+                    try {
+                        float factor = cordovaInstance.getActivity().getApplication().getBaseContext().getResources().getDisplayMetrics().density;
+                        width = (int)(options.getInt("width") * factor);
+                        height = (int)(options.getInt("height") * factor);
+
+                        final LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(width, height);
+                        layoutParams.gravity = Gravity.CENTER;
+
+                        listonicAd.setLayoutParams(layoutParams);
+
+                        callbackContext.success("Success!");
+
+                    } catch(JSONException e) {
+                        throw new IOError(e);
+                    }
+                }
+            });
+        }
+
+    }
+
+    private void show(JSONObject options, CallbackContext callbackContext) {
+        cordovaInstance.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                String zone = "";
+                String sex = "";
+                String age = "";
+                HashMap<String, String> map = new HashMap<>();
+
+                try {
+                    zone = options.getString("zone");
+                    sex = options.getString("sex");
+                    age = options.getString("age");
+                    map.put("sex", sex);
+                    map.put("age", age);
+                } catch(JSONException e) {
+                    throw new IOError(e);
```

Comment:
Tutaj nie powinno być coś w stylu `callbackContext.error("Error!");` czy rzucenie wyjątku to załatwia?

Reply:
będąc szczerym nie wiem czy to załatwia sprawę, ale nie spotkałem się z czymś takim w pluginie admoba